target/linux/armsr/image/Makefile

```diff
@@ -102,13 +103,7 @@ define Device/generic
   GRUB2_VARIANT := generic
   FILESYSTEMS := ext4 squashfs
   UBOOT := $(if $(CONFIG_aarch64),qemu_armv8,qemu_armv7)
-  DEVICE_PACKAGES += kmod-amazon-ena kmod-e1000e kmod-vmxnet3 kmod-rtc-rx8025 \
-	kmod-i2c-mux-pca954x kmod-gpio-pca953x partx-utils kmod-wdt-sp805 \
-	kmod-mvneta kmod-mvpp2 kmod-fsl-dpaa1-net kmod-fsl-dpaa2-net \
-	kmod-fsl-enetc-net kmod-dwmac-imx kmod-fsl-fec kmod-thunderx-net \
-	kmod-dwmac-rockchip kmod-dwmac-sun8i kmod-phy-aquantia kmod-phy-broadcom \
-	kmod-phy-marvell kmod-phy-marvell-10g kmod-sfp kmod-atlantic \
-	kmod-bcmgenet kmod-octeontx2-net
+  DEVICE_PACKAGES += partx-utils kmod
```

Comment:
> But what I know for sure is that OpenWrt would probably appreciate the effort since the packaging is sometimes lagging behind and proper kmod packages are not always engineered, so it would certainly be a nice and substantial contribution.

Its actually mandatory, this breaks for example purpose of [ImageBuilder](https://openwrt.org/docs/guide-user/additional-software/imagebuilder) and additional [tooling/frontends](https://openwrt.org/docs/guide-developer/imagebuilder_frontends) relying on this.

Since this target is not storage space constrained, I don't see a problem with having additional modules added to this list (ideally logically split into separate commits, so we can know/track/review which module is needed by specific armsr board). If some module is causing issues, then simply try to solve this upstream first and as a last resort we could use some downstream hack/patch if there is no other option/tradeoff left.

Reply:
For example https://firmware-selector.openwrt.org/?version=23.05.2&target=armsr%2Farmv8&id=generic

![image](https://github.com/openwrt/openwrt/assets/43848/47045d0a-5e3f-4357-bfe1-94a6a87a2277)

Reply:
You can customize the images by adding kmod-* packages. It's not broken. See https://github.com/openwrt/openwrt/pull/13684/commits/dad897b43446642477dc311f421cbb3c60050a3a

Reply:
BTW you can remove packages as well, see https://openwrt.org/docs/guide-user/additional-software/imagebuilder#selecting_packages

I would recommend to check [ImageBuilder out](https://github.com/openwrt/openwrt/blob/master/target/imagebuilder/files/Makefile), its quite easy:
 
```
podman run --rm -it \
    ghcr.io/openwrt/imagebuilder:armsr-armv8 sh -c \
    'make image PACKAGES="tmate -kmod-phy-marvell-10g"'
```

ImageBuilder basically downloads binary packages and creates images from those on the fly, so you're breaking this use case.

Reply:
I don't think this use case is broken. `opkg` works, see:

```
root@OpenWrt:/# opkg install kmod-nft-core_6.1.74-1_aarch64_generic.ipk
Installing kmod-nft-core (6.1.74-1) to root...
Configuring kmod-nft-core.
[ 1837.897514] kmodloader: loading kernel modules from /etc/modules.d/*
[ 1837.901291] kmodloader: done loading kernel modules from /etc/modules.d/*
```

I could also test the image builder further. I just need to know how to do it locally.